nilearn/maskers/tests/test_multi_nifti_maps_masker.py

```diff
@@ -129,6 +129,12 @@ def test_multi_nifti_maps_masker():
     masker.fit_transform(fmri22_img)
     np.testing.assert_array_equal(masker._resampled_maps_img_.affine, affine2)
 
+    # Test "MultiNiftiMapsMasker.fit_transform" with multiple "sample_mask"
+    sample_mask = np.arange(length - 2)
+    masker.fit_transform(
+        [fmri11_img, fmri11_img], sample_mask=[sample_mask, sample_mask]
+    )
+
```

Comment:
Quick comment: do you think you could extract this test in a separate test (in a separate function)? It would be also nice if you could refer to the github issue that that this test is related to: this will help anyone who reads the code know why the test is there.

This may require you to reuse some of the code that "sets up" the test.

See an example here: https://github.com/nilearn/nilearn/blob/93c49f1ff2219506411aa20c35a971288708261e/nilearn/plotting/tests/test_img_plotting/test_plot_stat_map.py#L85-L92

Other reasons why you want to have small independent test:
- if for some reason the test fails before it gets to the part you added, your new test never gets a chance to run
- by chaining tests like this, we add potential dependencies between tests (test B may reuse as input data that was already modified by test A)

To be fair, a lot of tests in our testing suite are written like long chains of tests so it does feel tempting to "add another one at the end", I am only complaining because I am spending quite a bit of time rewriting those tests to avoid this. :wink:

Reply:
Also I suspect that you may have some expectation of the "dimensions" of the result of `fit_transform`, it would be good to therefore add an assert statement that checks that the dimension of the output are correct.

Reply:
Sure, thanks for the suggestion - I admit that I blindly followed the structure of the file (first time contributing here :smile:).